src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -678,6 +678,17 @@ private void EmitReturnStatement(BoundReturnStatement boundReturnStatement)
                 }
                 else
                 {
+                    if (expressionOpt != null)
```

Comment:
The problem is that we have a *method* whose return type is an (untranslated) anonymous type. Normally there is no need to translate a method's return type.